src/imgui/ImGuiReverseBar.cc

```diff
@@ -184,28 +184,89 @@ void ImGuiReverseBar::showMenu(MSXMotherBoard* motherBoard)
 						}
 					});
 				}
-				ranges::sort(replayNames, std::greater<>{}, &ReplayNames::ftime);
 			}
 			if (replayNames.empty()) {
 				ImGui::TextUnformatted("No replays found"sv);
 			} else {
 				ImGui::TextUnformatted("Select replay"sv);
-				im::ListBox("##select-replay", [&]{
-					for (const auto& [fullName_, displayName_, _] : replayNames) {
-						const auto& fullName = fullName_; // clang workaround
-						const auto& displayName = displayName_; // clang workaround
-						if (ImGui::Selectable(displayName.c_str())) {
-							manager.executeDelayed(makeTclList("reverse", "loadreplay", fullName));
-						}
-						im::PopupContextItem([&]{
-							if (ImGui::MenuItem("delete")) {
-								confirmCmd = makeTclList("file", "delete", fullName);
-								confirmText = strCat("Delete replay '", displayName, "'?");
-								openConfirmPopup = true;
+				int flags = ImGuiTableFlags_RowBg |
+					ImGuiTableFlags_BordersV |
+					ImGuiTableFlags_BordersOuter |
+					ImGuiTableFlags_Resizable |
+					ImGuiTableFlags_Sortable |
+					ImGuiTableFlags_Hideable |
+					ImGuiTableFlags_Reorderable |
+					ImGuiTableFlags_ContextMenuInBody |
+					ImGuiTableFlags_SizingStretchProp;
+				im::Table("##select-replay", 2, flags, {}, [&]{
+					ImGui::TableSetupColumn("Name");
+					ImGui::TableSetupColumn("Date/time", ImGuiTableColumnFlags_DefaultSort | ImGuiTableColumnFlags_PreferSortDescending);
+					ImGui::TableHeadersRow();
+					// check sort order
+					auto* sortSpecs = ImGui::TableGetSortSpecs();
+					if (sortSpecs->SpecsDirty) {
+						sortSpecs->SpecsDirty = false;
+						assert(sortSpecs->SpecsCount == 1);
+						assert(sortSpecs->Specs);
+						assert(sortSpecs->Specs->SortOrder == 0);
+
+						switch (sortSpecs->Specs->ColumnIndex) {
+						case 0: // name
+							if (sortSpecs->Specs->SortDirection == ImGuiSortDirection_Descending) {
+								ranges::sort(replayNames, std::greater<>{}, &ReplayNames::displayName);
+							} else {
+								ranges::sort(replayNames, std::less<>{}, &ReplayNames::displayName);
 							}
-						});
+							break;
+						case 1: // time
+							if (sortSpecs->Specs->SortDirection == ImGuiSortDirection_Descending) {
+								ranges::sort(replayNames, std::greater<>{}, &ReplayNames::ftime);
+							} else {
+								ranges::sort(replayNames, std::less<>{}, &ReplayNames::ftime);
+							}
```

Comment:
Using our helper functions `sortUpDown_String()` and `sortUpDown_T()` may make it a bit less clunky.

Reply:
Thanks, that works well. I noticed the sorting was only working the first time, so I fixed that. Each time we recreate the list, it must also be sorted again, despite the dirty-flag saying it doesn't need to.

Reply:
I now noticed what I did to fix this isn't working. But I don't understand at all why.... can you help?

Reply:
I did some debugging. In the end I found that:
* I add a debug-print right before the call to `im::Table()` (around line 200), and one inside the lambda we pass to that function.
* Most of the time both prints trigger. _Except_ on the first frame that draws the sub-menu.
* But only on that first frame you have `loadReplayOpen == false`.

Speculation from my part: could it be that internally on the first frame ImGui only calculates the available size for the table, and then only on the next frames draws the content of the table??

So I'm afraid you'll have to add a workaround. Maybe add a `bool alreadySorted` that you reset to `false` when you re-populate `replayNames`?

Reply:
I think the current approach of re-sorting is fine. But just as a background remark (I needed this to convince myself it's fine):

In general when sorting columns in a table, you want to be able to first sort on column X, followed by sorting on column Y. More specifically, items that are equal in column Y should still remain sorted on column X. (That's why the `SortUpDown_xxx()` utility functions use _stable_-sort).
In principle when you re-populate the data, you have to keep this history in mind. Though in this specific case, because there won't be duplicates in the columns, it doesn't matter.

At least there won't be any duplicate names. In principle there could be duplicate times, but that should happen often, or at all.

Reply:
OK, I didn't think about it in this detail, but I did think it was good enough. Now it's also very clear why.